lib/zoom/params.rb

```diff
@@ -60,15 +60,42 @@ def hash_filter(filter)
       filter.keys
     end
 
-    def find_matching_keys(keys)
-      keys.flatten.each_with_object([]) do |key, array|
-        array << key if self[key]
+    def filter_required(filters)
+      # Unless value is a hash, filter
+      filters.each.with_object(self.class.new(except(filters.flatten))) do |filter, params|
+        case filter
+        when Symbol, String
+          params.delete(filter)
+        when Hash
+          filter.each do |k, v|
+            nested_filter = self.class.new(self[k]).filter_required(v)
+            if nested_filter.empty?
+              params.delete(k)
+            else
+              params[k] = nested_filter
+            end
+          end
+        end
       end
     end
 
-    def find_missing_keys(keys)
-      keys.each_with_object([]) do |k, array|
-        array << k if self[k].nil?
+    def find_missing_entries(*entries)
+      entries.flatten.each.with_object([]) do |entry, array|
+        if entry.is_a?(Hash)
+          entry.keys.each do |k|
+            array << k && next if self[k].nil?
+            missing_entries = self.class.new(self[k]).find_missing_entries(*entry[k])
+            array << { k => missing_entries } unless missing_entries.empty?
+          end
+        elsif self[entry].nil?
+          array << entry
+        end
+      end
+    end
```

Comment:
This section is kinda the meat & potatoes of this PR. Allows for requiring more complex data structures as defined in the API schema: https://marketplace.zoom.us/docs/api-reference/zoom-api/billing/accountplancreate

Reply:
Pretty hairy